lib/sgtn-client/api/source.rb

```diff
@@ -15,7 +19,7 @@ def self.getSource(component, key, locale)
         else
           SgtnClient.logger.debug "[Source][getSource]getting sources from cache with key: " + cache_key
         end
-        s = items.nil?? nil : items[locale][key]
+        s = (items.nil? || items[locale].nil?)? nil : items[locale][key]  
```

Comment:
This is not correct. Per spec, if source bundle is not found, then it should be loaded right there and then.
Also, in CloudHealth, loadBundles had been invoked at app start up.

Reply:
I don't understand the reason, maybe Xiaochao could help explain it. However, from the log, it's this line throw the exception.

Reply:
After adding this, what would the UI display then if no source strings is found?

Reply:
It's fixing issues report in the logs, it's saying the array is nil before using it:
undefined method `[]' for nil:NilClass
["/home/cp-web/cloudpercept/shared/bundle/ruby/2.5.0/gems/singleton-client-0.7.4/lib/sgtn-client/api/source.rb:18:in `getSource'"

Reply:
The array should not be nil in the first place.